### PR TITLE
Add separate sample app and test for Postgres

### LINKED ISSUE #1279: Add an integration test for the SQL sample with Postgres
Add an integration test for the SQL sample with Postgres. This would be useful to spot breakages like: #1277.

### PR DESCRIPTION
**Pull request by dzou**
This adds a separate sample app and test for Postgres. This serves the purpose of offering a simple example for postgres setup and also as an integration test for our postgres-related code.

Fixes #1279

### CODE REVIEW

File: spring-cloud-gcp-samples/spring-cloud-gcp-sql-postgres-sample/pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2017-2018 the original author or authors.
```
Comment (artembilan) [inline spring-cloud-gcp-samples/spring-cloud-gcp-sql-postgres-sample/pom.xml]: No Copyright in XML files, please.

File: spring-cloud-gcp-samples/pom.xml [hunk 2]
```diff
@@ -33,6 +33,7 @@
 		<module>spring-cloud-gcp-pubsub-binder-sample</module>
 		<module>spring-cloud-gcp-integration-pubsub-json-sample</module>
 		<module>spring-cloud-gcp-security-iap-sample</module>
+		<module>spring-cloud-gcp-sql-postgres-sample</module>
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/pom.xml]: Should we also rename `spring-cloud-gcp-sql-sample` to `spring-cloud-gcp-sql-mysql-sample`?

File: spring-cloud-gcp-samples/spring-cloud-gcp-sql-postgres-sample/README.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
+= Spring Cloud GCP PostgreSQL Sample
+
+This code sample demonstrates how to connect to a Postgres Google Cloud SQL instance using the link:../../spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql-postgresql[Spring Cloud GCP PostgreSQL Starter].
+
+You will create an instance, a database within the instance, populate the database and then query it.
+
+== Setup
+
+1. Follow https://cloud.google.com/sql/docs/mysql/quickstart[these instructions] to set up a Google Cloud Project with billing enabled.
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/spring-cloud-gcp-sql-postgres-sample/README.adoc]: mysql -> postgres?

File: spring-cloud-gcp-samples/spring-cloud-gcp-sql-postgres-sample/README.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
+= Spring Cloud GCP PostgreSQL Sample
+
+This code sample demonstrates how to connect to a Postgres Google Cloud SQL instance using the link:../../spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql-postgresql[Spring Cloud GCP PostgreSQL Starter].
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/spring-cloud-gcp-sql-postgres-sample/README.adoc]: Can we link to the MySQL sample here as well?

**Pull request merged**